prometheus/testutil/lint.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil/promlint"
+)
+
+// CollectAndLint registers the provided Collector with a newly created
+// pedantic Registry. It then does the same as GatherAndLint, gathering the
```

Comment:
```suggestion
// pedantic Registry. It then does the same as GatherAndLint, gathers the
```
Kind of scared to suggest this to you, but .... am I right here? :smile: 

Previous version was reading oddly

Reply:
I could argue why my version is correct. But I agree it's convoluted. I'll reword to avoid the complication.